packages/desktop-client/src/hooks/useSelected.tsx

```diff
@@ -274,9 +273,6 @@ export function SelectedProvider<T extends Item>({
 
   const dispatch = useCallback(
     async (action: Actions) => {
-      if (!action.event && isNonProductionEnvironment()) {
-        throw new Error('SelectedDispatch actions must have an event');
-      }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why are you changing this? Seems totally unrelated to the objective of this PR.

Reply:
To Select an item in the grid(i.e. have the found transaction pre selected) I need to dispatch the "Select" Action
This has historically only been done from an Event action, i.e. mouse click in the grid. 

As i need to call it from a useEffect(prior memoziation) this is called without an Event,
Hence this bit of code needs to be removed.

Reply:
Hmm.. do we need to update the selected items when creating a new schedule?

I'm really worried about what unintended consequences this particular change might have in other parts of the app, so ideally I would not like to see it changed if possible. If that means: the selected items get un-checked after creating a schedule - I think that's a good compromise.

Reply:
**Reason I don't want to change:**
If we don't change it then workflow wise its weird, we are either create a new schedule and adding a transaction to it without visualising that in the UI.
||
We are creating a schedule from a transaction, and then not assigning that transaction to the schedule by default. 

Of the two above if we 100% dont want to change this code i would go with optiion 2, 

**Reason I think we don't need to be worried about this change:**
<details>
  <summary>Current Check Occurs Here</summary>
`  const dispatch = useCallback(
    async (action: Actions) => {
    if (!action.event && isNonProductionEnvironment()) {
        throw new Error('SelectedDispatch actions must have an event');
      }
      if (action.type === 'select-all') {
        if (!action.event && isNonProductionEnvironment()) {
          throw new Error('SelectedDispatch actions must have an event');
        }
        if (latestItems.current && latestItems.current.size > 0) {
          return instance.dispatch({
            type: 'select-none',
            event: action.event,
          });
        } else {
          if (fetchAllIds) {
            return instance.dispatch({
              type: 'select-all',
              ids: await fetchAllIds(),
              event: action.event,
            });
          }
          return instance.dispatch({ type: 'select-all', event: action.event });
        }
      }
      return instance.dispatch(action);
    },
    [instance.dispatch, fetchAllIds],
  );` 
</details>
<details>
  <summary>Code Path A</summary>
`  return instance.dispatch({
            type: 'select-none',
            event: action.event,
          });`
`
        case 'select-none':
          return { ...state, selectedItems: new Set<string>() };
`
Event is unused. 
</details>
<details>
  <summary>Code Path B</summary>
`  if (fetchAllIds) {
            return instance.dispatch({
              type: 'select-all',
              ids: await fetchAllIds(),
              event: action.event,
            });
          }
`
`
        case 'select-all':
          let selectedItems: string[] = [];
          if (action.ids && items && selectAllFilter) {
            const idsToInclude = new Set(
              items.filter(selectAllFilter).map(item => item.id),
            );
            selectedItems = action.ids.filter(id => idsToInclude.has(id));
          } else if (items && selectAllFilter) {
            selectedItems = items.filter(selectAllFilter).map(item => item.id);
          } else {
            selectedItems = action.ids || items.map(item => item.id);
          }
          return {
            ...state,
            selectedItems: new Set(selectedItems),
            selectedRange:
              action.ids && action.ids.length === 1
                ? { start: action.ids[0], end: null }
                : null,
          };
`
Event is unused. 
</details>
<details>
  <summary>Code Path C</summary>
` 
return instance.dispatch({ type: 'select-all', event: action.event })
`
Same Path as Path B Event is unused. 
</details>
<details>
  <summary>Code Path D</summary>
` return instance.dispatch(action);
`
`
       case 'select': {
`
We have Added a new check to ensure event is correctly set. 
&&
`
default:
          throw new Error('Unexpected action: ' + JSON.stringify(action));
`
Event is unused. 
</details>

Finally, the calls into useSelected 
<details>
<summary> Details </summary>
1.  PayeeTableHeader -> passes Event
2. PayeeTableRow -> passes Event
3. RuleRow -> passes Event
4. RulesHeader -> passes Event
5. DiscoverSchedulesTable -> passes Event
6. SimpleTransactionsTable -> passes Event && New code which doesnt
7. selectAscDesc -> passes Event
8. onUpdateAfterConfirm -> passes Event

Reply:
Other option is we could create a "DummyEvent" but personal view is that is likely to cause bugs down the line.

Reply:
Ok, so I looked into this a bit more. It seems the underlying product problem we're trying to solve here is a lot more complicated actually. Apologies for continuing to send all these pieces of feedback, but I think that if we want to get this merged - it's important to get all the tiny details right so we wouldn't get bug reports about the feature.

So.. here's what I've noticed: the schedule creation modal does not respect the selected transactions.

https://github.com/actualbudget/actual/assets/886567/e230b938-236b-4d22-b26d-d28c90b3b930

We can think of the transaction table and schedule creation modal as two separate entities. The transaction table should pass the selected transaction ids to the creation modal. Next the user can interact with the schedule creation modal as usual: change some details AND select specific transactions in the list that is visible in the modal. We should preselect the transactions from the table. But the user can decide to un-select them there (which would only affect the modal, not the main table). And then they can click "create" which would link the new schedule with the transactions selected in the modal (not the table).

Lastly: we close the modal and "refresh" the state in the transaction table with the linked transactions. However, this should happen automatically without the need to dispatch any new event. You won't need any new logic for this.

One more thing to consider: if multiple transactions are selected - we cannot "blanket fill" the schedule creation modal. For multiple transactions it should use either the "one in" syntax or just leave the field blank (for example if different balances are selected).

Again: sorry for continuing to give feedback here. But I hope you see that the problem is a lot more complicated than we originally anticipated.

Reply:
Not sure I fully get you, 
If more than one transaction is initially selected then it will use the first one in the "create schedule" helper, happy to add a tooltip highlighting this, but believe this makes sense as functionally, it doesn't make sense to create a single scheduled transaction from multiple transactions.
As you say, if we look to support creating a schedule from multiple transaction, it gets very muddy very quickly, and i don't think it would be very useful. 

Rest of it seems to work correctly from what i can see. 
https://github.com/actualbudget/actual/assets/30550215/7b84d381-34ae-4e23-8458-437dab9f9db6

Reply:
Ok, let me clarify in one sentence: Inside the schedule creation modal/page only the selected transactions should be linked to the schedule after it is creatd.

Currently this is not the case. That's a bug. All the other feedback I provided should help you solve this bug.

<img width="763" alt="Screenshot 2024-01-21 at 17 00 33" src="https://github.com/actualbudget/actual/assets/886567/f9dce4a9-36cc-4083-be4b-e880f720273d">

Reply:
I agree with your statement, and as per below video, only transactions selected on the "Schedule Creation Model" are linked to the newly created schedule. I can't see the bug you are detailing. 

See video:
https://github.com/actualbudget/actual/assets/30550215/d10b1eb5-69e8-457e-a3f6-722a98673ef7

Reply:
1. one selected transaction, but later un-selected: yet it still gets attached to the schedule


https://github.com/actualbudget/actual/assets/886567/a33e6630-5111-44ce-b3cc-dc31b018502f


2. two selected transactions: only a single showing up in the modal


https://github.com/actualbudget/actual/assets/886567/5bac5258-d6ff-4e35-adc1-236b71fb3502


3. single transaction selected, but later the schedule config changed and multiple transactions selected: neither gets linked to the schedule


https://github.com/actualbudget/actual/assets/886567/081dff5e-984e-4819-ba60-d5b63b937ffe

Reply:
ahah, you are correct, 
needed a fresh set of my eyes. 
Adjusted to fix and this cleans up the other comments you had as well. 
Let me know thoughts.

Reply:
Adjusted it so the transaction you initially selected is always visible in the search trans box as well. My view, would be odd for you to select the tran to create from, but then not link it. 
Happy to remove if you disagree.

Reply:
Apologies for the slow reply; I've been on vacation.

-

I'm still able to reproduce most of [these bugs](https://github.com/actualbudget/actual/pull/2222#discussion_r1461031031). Your work here is definitely very much appreciated and I don't want to sound like I'm attacking your work, but I'd strongly recommend re-reading [this comment](https://github.com/actualbudget/actual/pull/2222#discussion_r1460976171). It encapsulates the idea how you could make the code cleaner while also fixing the bugs. Hope that helps!

Reply:
Hope you had a good vacation, 
No Problem on your response, Think I need to also explain myself better :). 

Going through the 
bugs you highlighted:
1) one selected transaction, but later un-selected: yet it still gets attached to the schedule - **Resolved**

https://github.com/actualbudget/actual/assets/30550215/7b02b25d-3b2d-46e0-809f-89f8ab81d40a

2) two selected transactions: only a single showing up in the modal - **Bypassed**
When selecting multiple transactions to create a new schedule from as you rightly highlighted it becomes messy & complex very quickly. Given in my view 95% of the time people are going to use this feature it will be by selecting a single transaction I have only implemented it to support this. As per comment here:

> If more than one transaction is initially selected then it will use the first one in the "create schedule" helper, happy to add a tooltip highlighting this, but believe this makes sense as functionally, it doesn't make sense to create a single scheduled transaction from multiple transactions.

I have updated this now, so the create option is only displayed when a single transaction is selected to simplify this, and avoid confusion from users. - See Video below **Happy to adjust this to disable the button instead with an explanation tooltip if you think that would be more suitable, I thought simply hiding the button looked cleaner**

https://github.com/actualbudget/actual/assets/30550215/6cea4ba8-44dc-4d3b-b9a5-f74009898ae0

3) This bug looks to be 2 different existing bugs you have triggered at the same time,  - **Existing Bug**

3a) When you change details in the add schedule model, previous selections are removed - See Video below highlighting it in the add schedule page. This should be fairly straight forward to fix but would make sense to do in another PR, happy to do this change. 

https://github.com/actualbudget/actual/assets/30550215/47cb17f2-ba5c-4bef-8b4b-0e650c155b45

3b) When a schedule is created and items are linked to it, if you toggle the select all on the transaction, the transaction seems to go into a "bad state" and even though the schedule shows the transactions as linked, the transactions themselves don't show as linked. 
Again, this looks to be an existing bug, that I am more than happy to tackle as a new PR. 
See video

https://github.com/actualbudget/actual/assets/30550215/ec58da0f-5404-4a4f-8530-cbc0dcbfff68

Reply:
1 - patched
2 - patched
3 - patched - but I think you misunderstood the bug report there. The issue was that when multiple transactions are selected in the modal (via select-all button) - neither of them would get linked to the schedule after the schedule creation.

The existing behaviour of resetting the selected transactions (in the modal) after the search parameters change is working as-expected. No need to change that.